### ISSUE TITLE
liveFeed videos don't work

### ISSUE THREAD

**Issue by mrcalbert** [rrirower/DP2SceneGraphWizard#111]
After adding a liveFeed video and trying to play, it will crash Roku completely and require a reboot

For me it fails with this

Warning occurred while setting a field of an RoSGNode
-- Type mismatch occurred when setting the "focusedChild" field of a "ContentNode" node
pkg:/components/UILogic/VideoPlayer.brs(171)

**Comment by rrirower** [comment]
Please provide a valid feed URL that contains a live feed for testing. 

Please use the bug report template to report issues.

**Comment by rrirower** [comment]
I can't help you without the info I've asked for.

**Comment by mrcalbert** [comment]
Apologies! I'm rather fresh to Github and didn't see the notification that you responded last week. I believe this is the template:

**Describe the bug**
Error loading videos categorized as liveFeeds. I've tried livestreams from my company and test feeds online. I'm currently having difficulties finding a non branded free test stream, however this sample one breaks as well

https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8

Note that the same video URLS entered as an episode in a series works fine.

**To Reproduce**
1. Create a "liveFeed" in the Channel Feed Manager with the required info. Enter a HLS stream
2. Once you upload the json and see it on your channel, try to watch this live video
3. Video player launches and you see the "loading circle" for .5 seconds and Roku device crashes

**Expected behavior**
Hoping to play a live video or an HLS video under the liveFeed. The same URL works as an episode in a series however that's a bit nested to go into for a quick live stream video.

**Screenshots**
N/A

**Version number:**
Roku Ultra
Version: 13.0.0 build 24056-C2
Just updated today however had issue before on previous version

**Additional context**
Brightscript Debugger through Visual Studio Code seems to report the error below:

=================================================================
Warning occurred while setting a field of an RoSGNode
-- Type mismatch occurred when setting the "change" field of a "ContentNode" node
pkg:/components/UILogic/VideoPlayer.brs(171)
=================================================================
=================================================================
Warning occurred while setting a field of an RoSGNode
-- Type mismatch occurred when setting the "focusedChild" field of a "ContentNode" node
pkg:/components/UILogic/VideoPlayer.brs(171)

Thank you!

**Comment by rrirower** [comment]
Thanks for your response.  However, as stated above...

`Please provide a valid feed URL that contains a live feed for testing.`

I need the URL to your feed file, not the URL to the live stream.  The URL to the live stream would be contained in your feed file.

**Comment by samadtehrani** [comment]
Hi,
I have the same problem with my live channel. This is my test feed: http://10674-5.s.cdn13.com/Mihan_live_test.json

**Comment by rrirower** [comment]
@samadtehrani thanks for your feedback. I will look into this now that I have a test case.

**Comment by rrirower** [comment]
@samadtehrani Is the m3u8 url in your feed file valid?  I tried to play it in a web browser and it threw errors.

**Comment by rrirower** [comment]
@samadtehrani I did get the video to play in the stream tester.

**Comment by samadtehrani** [comment]
Yes, It is valid. You cannot play .m3u8 playlists in the browser but every video player like VLC or Roku's player can play.

**Comment by ferdiworks** [comment]
@samadtehrani You can play M3U8 in chrome with the chrome plugin! However your link https://iptv.mihantv.com/mihantv/playlist.m3u8 does not play!

**Comment by ferdiworks** [comment]
@samadtehrani I noticed your livefeed link https://iptv.mihantv.com/mihantv/playlist.m3u8 is from IPTV and may be blocked due to copyright or security issues!

**Comment by samadtehrani** [comment]
Looks like you guys are missing the point. Live channels do not work, It restarts the Roku. no matter what livestream you replace with the sample I added to the feed. 
I added these free samples that are available online, Same result.
http://sample.vodobox.net/skate_phantom_flex_4k/skate_phantom_flex_4k.m3u8

http://playertest.longtailvideo.com/adaptive/wowzaid3/playlist.m3u8

http://cdn-fms.rbs.com.br/vod/hls_sample1_manifest.m3u8

http://nasatv-lh.akamaihd.net/i/NASA_101@319270/index_1000_av-p.m3u8?sd=10&rebase=on

http://content.jwplatform.com/manifests/vM7nH0Kl.m3u8

http://walterebert.com/playground/video/hls/sintel-trailer.m3u8

http://qthttp.apple.com.edgesuite.net/1010qwoeiuryfg/sl.m3u8

https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_16x9/bipbop_16x9_variant.m3u8

**Comment by rrirower** [comment]
@ferdiworks @jsonnentag Do you guys know of a live stream that works?  I'm trying to debug this, but, I'm not getting too far.

**Comment by AndrewRCD** [comment]
I manage some streams for clients. If you need one for testing, is there a way to send you the link privately?

**Comment by rrirower** [comment]
@AndrewRCD  Actually, if you can tell me that you've got a channel that uses Live Feeds (HLS) and is working, that's good enough.  I need to know if this is a problem for everyone or just a few.  If for a few, I know that streaming services can be picky about security, etc.

**Comment by AndrewRCD** [comment]
I do :) I use both normal HLS and low-latency HLS ( LL-HLS) for videos, and that works too

**Comment by rrirower** [comment]
@AndrewRCD Thanks.  Were those channels generated using the Wizard tool?

**Comment by AndrewRCD** [comment]
yep

**Comment by rrirower** [comment]
@AndrewRCD  Thanks again.  I hope that those seeing your reply will take a closer look at the streaming service they are using.

**Comment by rrirower** [comment]
@AndrewRCD Lastly, did you make any changes to the Brightscript code to get the Live stream working?

**Comment by AndrewRCD** [comment]
No - it worked "out of the box" and no changes or adaptations of any kind were needed.

**Comment by samadtehrani** [comment]
@AndrewRCD And do you use the latest version at this point "Version 1.0.0.2"?

**Comment by AndrewRCD** [comment]
yes @samadtehrani every time the wizard is updated I update the channel with the latest version just to make sure that I have the latest code

**Comment by AndrewRCD** [comment]
I am using both tools since the wizard tool was in beta - and I don't have any issues with my live streams or my videos

**Comment by samadtehrani** [comment]
Strange. I used the code "out of the box" a few months ago and it worked with the same condition. Now with Version 1.0.0.2 it freezes and restarts the Roku!

**Comment by rrirower** [comment]
@samadtehrani I get the same symptoms when I test with your feed file.  I would suggest taking a closer look at your m3u8 file and/or the service you are using.

**Comment by AndrewRCD** [comment]
I don't know what to say to that, for me, it is working, maybe something with your feed settings -

**Comment by AndrewRCD** [comment]
@samadtehrani I noticed in your content feed some ....  things i most certainly wouldn't use like that 
for example 
    "captions": [
                    {
                        "url": "https://",
                        "language": "en",
                        "captionType": "SUBTITLE"

but you dont have a URL  - then why have it at all in the code ....

the other 
 "adBreaks": [
                    "00:00:00"
                ]

this also for me makes no sense 

maybe it is nothing but ...  why complicate a code is not needed :)

**Comment by samadtehrani** [comment]
@AndrewRCD Are you sure that you are using Channel Feed Manager to produce your feed? Because you have to add a caption and at least one adBreak to make your feed. 
Another question: Do you add your media as "livefeed"? you know that we can add live channels as shortvideo format or anything else and it is working.

**Comment by AndrewRCD** [comment]
actually, you don't have to - subtitles are not mandatory yet - and for live streams, it will not be, those make exceptions - and ad add breaks are also not mandatory

**Comment by samadtehrani** [comment]
You mean you don't get this message on "Channel Feed Manager"? 
![Screenshot 2024-05-28 135218](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/4719588c-d920-4a2a-b788-9d8d0459acb9)

**Comment by ferdiworks** [comment]
We have a few clients that have livestreaming issues w Roku. Two are using the software from here. After talking with support here are some things reported to check on our own side:

Live streaming to Roku and other OTT services

Your encoder settings for livestreaming should be checked as Roku and other providers need certain settings!

You need a video to stream at 1920x1080
set the video to 1920x1080 and if your setting has a rescale output: change the VIDEO to 1920x1080 also!

Next change rescale output to disabled!

change or set the bitrate to 3370kbps or 4000

If you are streaming from your desktop:
Your internet service should be consistently 2x higher than the output bitrate or the stream will suffer. You can check with https://testmy.net/upload

**Comment by ferdiworks** [comment]
![thumbnail](https://github.com/rrirower/DP2SceneGraphWizard/assets/40750492/0f844777-9fe8-42b0-9581-e4a0ba95ca9e)
The Error is a 404 etra:etype:buffer

**Comment by ferdiworks** [comment]
Note I found its not just our use of this software and see issues with Livestreaming from outside feeds and services!

**Comment by samadtehrani** [comment]
I replaced my m3u8 with several livefeeds. same results. I even used Big Buck Bunny (https://live-par-2-cdn-alt.livepush.io/live/bigbuckbunnyclip/index.m3u8) which Roku uses for testing on the Roku Platform or (https://bitdash-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8) that Roku uses for testing purposes. Same result.

**Comment by rrirower** [comment]
Closed.  No additional user feedback.
